src/drive/web/modules/views/OnlyOffice/Toolbar/useFileWithPath.jsx

```diff
@@ -0,0 +1,25 @@
+import { isQueryLoading, useQuery } from 'cozy-client'
+
+import { buildFileByIdQuery } from 'drive/web/modules/queries'
+
+export const useFileWithPath = fileId => {
+  const fileQuery = buildFileByIdQuery(fileId)
+  const fileResult = useQuery(fileQuery.definition, fileQuery.options)
+  const resultData = fileResult.data
+  const dirId = resultData ? resultData.dir_id : {}
+
+  const parentQuery = buildFileByIdQuery(dirId)
+  const parentResult = useQuery(parentQuery.definition, parentQuery.options)
+  const parentData = parentResult.data
+
+  if (isQueryLoading(fileResult) || isQueryLoading(parentResult)) {
+    return { data: null }
+  }
+
+  return {
+    data: {
+      ...resultData,
+      displayedPath: parentData ? parentData.path : undefined
+    }
+  }
+}
```

Comment:
We already have something similar with useFilesQueryWithPath. Why not reusing it?

Reply:
`useFilesQueryWithPath` ne prend pas en charge un fichier seul, et utiliser un query qui `getByIds` ce que la stack ne permet pas de faire en vue publique. Je fais refacto pour centraliser les 2 méthodes et ajouter une explication.